modules/fbs/includes/fbs.debt.inc

```diff
@@ -51,24 +52,41 @@ function fbs_debt_list($account) {
       'type' => $fee->reasonMessage,
     );
 
-    if (!empty($fee->materials)) {
-      $record_ids = array();
-      $material_ids = array();
-      foreach ($fee->materials as $material) {
-        $material_ids[] = $material->materialItemNumber;
-        $record_ids[] = $material->recordId;
+    // If the has associated materials; collect some information about them and
+    // try to build a more useful title.
+    $materials= array();
+    foreach ($fee->materials as $material) {
+      $ding_entity_id = $ids[$material->recordId];
+      if ($ding_entity_id && $entity = ding_entity_load($ding_entity_id)) {
```

Comment:
Could the entities loaded in line 37 not be used here. The function "ding_entity_load_multiple" returns the loaded entities and it seams to be the same function the code is in.

Reply:
Your right they could. No reason to load them again even though they are cached. I'll update the code tomorrow.

Reply:
I looked into this and on second thought we should probably just keep it as is.

The problem is that the entities returned from ding_entity_load_multiple() is keyed by their local id (tid) from the database. So we would have to do something like this:

```
$entities = ding_entity_load_multiple(array_filter($ids));
...
foreach ($fee->materials as $material) {
      $ding_entity_id = $ids[$material->recordId];
      $key = array_search(
        $ding_entity_id,
        array_column(get_object_vars($entities), 'ding_entity_id')
      );
      if ($key) {
        $entity = $entities[$key];
        // Do something with the ent
      }
```
Or else we could loop through allt the entities at once and make a new array keyed by ding_entity_id.

In any case, it's more messy than before. And considering that all the entities gets statically cached as a result of the call to ding_entity_load_multiple(), I think the best approach is to just load them one by one with ding_entitye_load().

Reply:
I guess another more compact solultion for making the mapping array would be:

```
$entities = ding_entity_load_multiple(array_filter($ids));
$entities_map = array_flip(array_column(get_object_vars($entities), 'ding_entity_id'));
```

Then we could easily get the key to use on $entities arrray:
```
foreach ($fee->materials as $material) {
  $ding_entity_id = $ids[$material->recordId];
  if (isset($entities[$entities_map[$ding_entity_id]]) {
    $entity = $entities[$entities_map[$ding_entity_id]];
  }
```

Let me know if I should update the code with this approach or just leave it like it is now.

Reply:
Forget what i wrote above. I found a more elegant solution and just made it :)